karapace/kafka_rest_apis/__init__.py

```diff
@@ -159,6 +167,9 @@ def _add_kafka_rest_routes(self) -> None:
         self.route("/topics/<topic:path>", callback=self.topic_publish, method="POST", rest_request=True)
 
     async def _maybe_create_async_producer(self) -> AIOKafkaProducer:
+        if self._async_producer is not None:
+            return self._async_producer
+
```

Comment:
Is this just an optimization? After all if `_async_producer is None`, the `while` loop below is never executed and `self._async_producer` is returned.

OR

Is this to avoid using the lock. If so, why?

Reply:
Avoid using the lock as this is feasible optimization route. The `_async_producer` is created one, never set back to `None`.